src/main/java/seedu/address/logic/commands/ArchiveCommand.java

```diff
@@ -0,0 +1,198 @@
+
+        return new CommandResult(String.format(
+                MESSAGE_ARCHIVED_DELIVERY_SUCCESS,
+                inspectedPerson.getName(),
+                Messages.formatDeliveryList(reverseList(deliveryToArchiveList))),
+                DeliveryAction.ARCHIVE);
+    }
+
+    /**
+     * Validates the indexes in the indexList to ensure none are out of bounds or duplicates.
+     *
+     * @param listSize The size of the list from which items are to be deleted.
+     * @param indexList The list of indexes to be validated.
+     * @throws CommandException if any index is out of bounds or if duplicates are found.
+     */
+    private void validateIndexes(int listSize, List<Index> indexList) throws CommandException {
+        boolean duplicate = hasDuplicates(indexList);
+        for (Index targetIndex : indexList) {
+            if (targetIndex.getZeroBased() >= listSize || duplicate) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+        }
+    }
+
+    /**
+     * Archive deliveries from the inspected person's delivery list based on the provided indexList.
+     *
+     * @param inspectedPerson The person whose deliveries are to be archived.
+     * @param deliveryList The list of deliveries to archive from.
+     * @return A list of deliveries that were archived.
+     */
+    private List<Delivery> archiveDeliveries(Person inspectedPerson, List<Delivery> deliveryList) {
+        List<Delivery> deliveryToArchiveList = new ArrayList<>();
+        for (Index targetIndex : indexList) {
+            Delivery deliveryToArchive = deliveryList.get(targetIndex.getZeroBased());
+            Delivery archivedDelivery = createArchivedDelivery(deliveryToArchive);
+            inspectedPerson.setDelivery(deliveryToArchive, archivedDelivery);
+        }
+        return deliveryToArchiveList;
+    }
+
+    /**
+     * Creates and returns a {@code Delivery} with the details of {@code toEdit}
+     * edited with {@code descriptor}.
+     */
+    private static Delivery createArchivedDelivery(Delivery toArchive) {
+        assert toArchive != null;
+
+        ItemName itemName = toArchive.getItemName();
+        Address updatedAddress = toArchive.getAddress();
+        Cost updatedCost = toArchive.getCost();
+        Eta updatedEta = toArchive.getEta();
+        Status updatedStatus = toArchive.getStatus();
+
+        Archive archive = new Archive(true);
+
+        return new Delivery(itemName, updatedAddress, updatedCost, updatedEta, updatedStatus, archive);
+    }
+
+    /**
+     * Reverses the order of the provided list.
+     *
+     * @param <T> The type of the elements in the list.
+     * @param list The list to be reversed.
+     * @return A reversed copy of the provided list.
+     */
+    private <T> List<T> reverseList(List<T> list) {
+        List<T> reversedList = new ArrayList<>(list);
+        Collections.reverse(reversedList);
+        return reversedList;
+    }
```

Comment:
Is there a reason why this method is needed instead of calling `Collections.reverse` directly on deliveryToArchiveList on Line 81? 🤔